pkg/cluster/k8sres.go

```diff
@@ -599,6 +599,22 @@ func generatePodAntiAffinity(podAffinityTerm v1.PodAffinityTerm, preferredDuring
 	return podAntiAffinity
 }
 
+func generateTopologySpreadConstraints(labels labels.Set, additionalTopologySpreadConstraints []v1.TopologySpreadConstraint) []v1.TopologySpreadConstraint {
```

Comment:
Help me to understand what this function is doing:
- Do we have to define this first hard-coded TopologySpreadConstraint when somebody specifies constraints in the manifest?
- What would happen if it is missing?
- Should the operator always create this spread constraint, similar to the node affinities?

Reply:
The purpose of this function is:

- If [`topologySpreadConstraints`](https://github.com/laiminhtrung1997/postgres-operator/blob/fbac974b3d33a9bb3f07c3c249a3b7c68fccfa1b/pkg/cluster/k8sres.go#L840) is set to `true` and [`additionalTopologySpreadConstraints`](https://github.com/laiminhtrung1997/postgres-operator/blob/fbac974b3d33a9bb3f07c3c249a3b7c68fccfa1b/pkg/cluster/k8sres.go#L841) is either empty or undefined, the operator will apply default constraints as hardcoded.
- If `additionalTopologySpreadConstraints` is defined, the specified list of constraints will be appended.
- The `topologySpreadConstraints` setting is configured to make the constraints customizable.

Reply:
@FxKu 
Our logic might be different. Please let me know how you’d like the operator to apply the constraint, and I’ll implement it according to your suggestion. I'd love to do it.

Reply:
@laiminhtrung1997 you have only described what the code does. I can read and understand code myself :smiley: 

Please try to answer my questions. I'm wondering if the function is needed at all? Why not go with what people specify in the manifest? I should have made this thought more clear. Hope you will understand my questions better now.

Reply:
@FxKu 

Okay, I got it. So, the only configuration input by users is topologySpreadConstraints, which will be generated in the manifest.